include/battery_monitor.h

```diff
@@ -0,0 +1,70 @@
+#ifndef BATTERYMONITOR_H
+#define BATTERYMONITOR_H
+
+#include "analog_pin.h"
+#include "system.h"
+
+class BatteryMonitor
+{
+public:
+  /**
+   * @brief Initialize the battery monitor.
+   * @details Does not initialize the ADC. The ADC must be initialized before calling.
+   * @param def The hardware definition struct for the battery monitor, mostly pin info
+   * @param adc The ADC used for the monitor. Must be initialized already
+   * @param voltage_multiplier The voltage multiplier. Set to 0 for no voltage sense.
+   * See set_voltage_multiplier.
+   * @param current_multiplier The current multiplier. Set to 0 for no current sense.
+   * See set_current_multiplier
+   */
+  void init(const BatteryMonitorHardwareStruct &def, AnalogDigitalConverter *adc, float voltage_multiplier=0, float current_multiplier=0);
+  /**
+   * @brief Read the current battery voltage
+   * @return The current battery voltage, or 0 if voltage sense is not available
+   */
+  float read_voltage() const;
+  /**
+   * @brief Read the current battery current
+   * @return The current battery current, or 0 if current sense is not available
+   */
+  float read_current() const;
+  /**
+   * @brief Sets the voltage multiplier for the battery monitor.
+   * @details Because the battery voltage is usually much too high for the ADC to read,
+   * analog battery monitors use a voltage divider or similar to reduce the voltage to a
+   * reasonable level. The voltage read by the ADC is multiplied by the voltage multiplier
+   * to get the actual reading. This number can be obtained from the spec sheet for the
+   * battery monitor, or through simple math for DIY monitors. This value is in volts per volt
+   * (i.e. unitless).
```

Comment:
could we have a little example formula here?

i.e

output = multiplier * observed voltage